tools/api_proto_breaking_change_detector/detector.py

```diff
@@ -13,31 +13,29 @@
 The tool is currently implemented with buf (https://buf.build/)
 """
 
-import tempfile
-from rules_python.python.runfiles import runfiles
-from tools.run_command import run_command
-from shutil import copyfile
+from buf_utils import check_breaking, make_lock, pull_buf_deps
 from pathlib import Path
-import os
 from typing import List
+from detector_errors import ChangeDetectorError
 
 
 class ProtoBreakingChangeDetector(object):
     """Abstract breaking change detector interface"""
 
-    def __init__(self, path_to_before: str, path_to_after: str) -> None:
+    def __init__(self, path_to_lock_file: str, path_to_changed_dir: str) -> None:
         """Initialize the configuration of the breaking change detector
 
         This function sets up any necessary config without actually
         running the detector against any proto files.
 
-        Takes in a single protobuf as 2 files, in a ``before`` state
-        and an ``after`` state, and checks if the ``after`` state
-        violates any breaking change rules.
+        Takes in a path to a lock file representing a proto "snapshot"
+        of the state before changes, and a path to a directory of proto
+        files that have been changed, and checks if the changes violate
+        any breaking change rules.
 
         Args:
-            path_to_before {str} -- absolute path to the .proto file in the before state
-            path_to_after {str} -- absolute path to the .proto file in the after state
+            path_to_lock_file {str} -- absolute path to the lock file representing the before state
+            path_to_changed_dir {str} -- absolute path to a directory containing proto files in the after state
         """
         pass
```

Comment:
It looks like this __init__ is completely unused and isn't called below. If that's intentional we should get rid of it.